src/db/entities/user.ts

```diff
@@ -0,0 +1,9 @@
+import { Column, Entity, PrimaryGeneratedColumn } from "typeorm"
+
+@Entity('user')
+export class User {
+  @PrimaryGeneratedColumn('increment')
+  id!: number
```

Comment:
O que você quis dizer com o `!`? Se for pra dizer que é um campo obrigatório, o typeORM já considera por padrão  como obrigatório os campos da entidade, a não ser que você declare eles como `nullable: true`

Reply:
Isso é por conta do TS, pois não é gerado no código transpilado. 
Daí intencionalmente em outras partes do código essa configuração do TS permite verificar partes em que é possível ser nulo.
Logo, se eu eliminar o `!` ele vai marcar indicar um erro sem transpilar. Mas se eu configurar para desconsiderar o `null` no projeto pode não ser uma melhor ideia.